controllers/servers/host_definer/watcher/csi_node_watcher.py

```diff
@@ -0,0 +1,86 @@
+import time
+from threading import Thread
+
+from controllers.common.csi_logger import get_stdout_logger
+from controllers.servers.host_definer.watcher.watcher_helper import Watcher, NODES, SECRET_IDS
+from controllers.servers.host_definer import settings
+
+logger = get_stdout_logger()
+
+
+class CsiNodeWatcher(Watcher):
+
+    def add_initial_csi_nodes(self):
+        csi_nodes = self._get_csi_nodes_with_driver()
+        for csi_node in csi_nodes:
+            if self._is_host_can_be_defined(csi_node.name):
+                self._add_node_to_nodes(csi_node)
+
+    def watch_csi_nodes_resources(self):
+        while True:
+            resource_version = self.csi_nodes_api.get().metadata.resourceVersion
+            stream = self.csi_nodes_api.watch(resource_version=resource_version, timeout=5)
+            for event in stream:
+                csi_node = self._get_csi_node_object(event[settings.OBJECT_KEY])
+                if (event[settings.TYPE_KEY] == settings.DELETED_EVENT) and (csi_node.name in NODES):
+                    self._handle_deleted_csi_node_pod(csi_node.name)
+                elif event[settings.TYPE_KEY] == settings.MODIFIED_EVENT:
```

Comment:
```suggestion
                csi_node = self._get_csi_node_object(event.object)
                if (event.type == settings.DELETED_EVENT) and (csi_node.name in NODES):
                    self._handle_deleted_csi_node_pod(csi_node.name)
                elif event.type == settings.MODIFIED_EVENT:
```
and if this doesn't work, consider converting the event to a munch or something